src/Symfony/Component/Cache/Traits/PdoTrait.php

```diff
@@ -140,6 +140,33 @@ public function createTable()
     /**
      * {@inheritdoc}
      */
+    public function prune()
+    {
+        $pdoConn = $this->getConnection();
+        $selects = $pdoConn->prepare("SELECT $this->idCol FROM $this->table WHERE $this->lifetimeCol + $this->timeCol <= :time");
```

Comment:
This implementation is no-go, it doesn't scale at all. Can't we just do one single DELETE instead?